Dockerfile

```diff
@@ -1,62 +1,79 @@
-FROM apache/zeppelin:0.7.3
+FROM apache/zeppelin:0.9.0
 
 MAINTAINER Saagie
 
-# Install Spark 2.1.0 for Hadoop 2.6
-RUN cd /tmp && wget https://archive.apache.org/dist/spark/spark-2.1.0/spark-2.1.0-bin-hadoop2.6.tgz -O /tmp/spark-2.1.0-bin-hadoop2.6.tgz
+ENV DEBIAN_FRONTEND noninteractive
+
+ENV JAVA_VERSION 8.131
+ENV SPARK_VERSION 2.4.5
+ENV HADOOP_VERSION 2.6
+ENV MESOS_VERSION 1.3.1
+ENV MESOS_VERSION2 2.0.1
+
+ENV HADOOP_HOME /etc/hadoop
+ENV HADOOP_CONF_DIR /etc/hadoop/conf
+ENV SPARK_HOME /usr/local/spark/${SPARK_VERSION}
+
+ENV MESOS_NATIVE_JAVA_LIBRARY /usr/lib/libmesos-${MESOS_VERSION}.so
+
+ENV ZEPPELIN_NOTEBOOK_DIR '/notebook'
+ENV ZEPPELIN_INTP_CLASSPATH_OVERRIDES /etc/hive/conf
+
+
+########################## LIBS PART BEGIN ##########################
+USER root
+# FIXME Remove once apache/zeppelin image will include PR :
+# https://github.com/apache/zeppelin/pull/3755
+# correcting issue :  https://issues.apache.org/jira/browse/ZEPPELIN-4783
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 51716619E084DAB9 && \
+    apt-get update -qq && apt-get install -yqq --no-install-recommends \
+      jq \
+      vim \
+    && rm -rf /var/lib/apt/lists/*
+########################## LIBS PART END ##########################
+
+
+########################## Install Spark BEGIN ##########################
+RUN wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz -P /tmp \
+    && tar -zxf /tmp/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz -C /usr/local/  \
+    && ln -s /usr/local/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}/ /usr/local/spark \
+    && ln -s /usr/local/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}/ /usr/local/spark2 \
+    && rm /tmp/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz
```

Comment:
Is the copy to **/usr/local/spark2** really needed? I remember we did that for Jupyter Spark but can't remember why.

Reply:
I too have no idea why it was there but as I'm not supposed to change the image structure I let it like it was before.
If you are sure I can remove it, so I'll.

Reply:
Ok this is a typo, in fact I spark2 was supposed to be spark/2.4.5 as necessary ...